src/checkstyle/checks/Check.hx

```diff
@@ -12,17 +12,25 @@ class Check {
 
 	public var severity:SeverityLevel;
 	public var type(default, null):CheckType;
+	public var categories:Array<String>;
+	public var points:Int;
+	public var desc:String;
 
-	var messages:Array<LintMessage>;
+	var messages:Array<CheckMessage>;
 	var moduleName:String;
 	var checker:Checker;
 
+	// Categories: Bug Risk, Clarity, Compatibility, Complexity, Duplication, Performance, Security, Style
```

Comment:
I didn't mean adding a category named "enum", but creating an enum for the categories so you don't have string literals everywhere. :)